src/backend/cdb/cdbllize.c

```diff
@@ -453,6 +454,24 @@ ParallelizeCorrelatedSubPlanMutator(Node *node, ParallelizeCorrelatedPlanWalkerC
 						 errdetail("In a subquery FROM clause, a function invocation cannot contain a correlated reference.")));
 			}
 		}
+		/* Check whether FunctionScan can be materialized
+		   FunctionScans with locus Entry can be materialized and
+		   broadcased if their paramenters are not depending on outer
+		   query results */
+		if (((Plan *)node)->flow->locustype == CdbLocusType_Entry &&
+			ctx->movement == MOVEMENT_BROADCAST)
```

Comment:
Why not move these conditions higher before the error and set `funcScanCanBeMaterialized` depending on them.